tests/framework/metrics/test_fscores.py

```diff
@@ -150,3 +170,33 @@ def _compute(self, *args, **kwargs):
         fbeta = FBeta(*args, **kwargs)
         fbeta(self.predictions, self.targets)
         return fbeta.get_metric()
+
+    def test_names(self):
+        fbeta = FBeta(average='macro')
+        self.assertEqual(['fscore_macro', 'precision_macro', 'recall_macro'], fbeta.__name__)
+        fbeta = FBeta(average='micro')
+        self.assertEqual(['fscore_micro', 'precision_micro', 'recall_micro'], fbeta.__name__)
+        fbeta = FBeta(average='micro', names=['f', 'p', 'r'])
+        self.assertEqual(['f', 'p', 'r'], fbeta.__name__)
+        fbeta = FBeta(average=0)
+        self.assertEqual(['fscore_0', 'precision_0', 'recall_0'], fbeta.__name__)
+        fbeta = FBeta(metric='fscore', average='macro')
+        self.assertEqual('fscore_macro', fbeta.__name__)
+        fbeta = FBeta(metric='fscore', average='micro')
+        self.assertEqual('fscore_micro', fbeta.__name__)
+        fbeta = FBeta(metric='fscore', average=0)
+        self.assertEqual('fscore_0', fbeta.__name__)
+        fbeta = FBeta(metric='precision', average='macro')
+        self.assertEqual('precision_macro', fbeta.__name__)
+        fbeta = FBeta(metric='precision', average='micro')
+        self.assertEqual('precision_micro', fbeta.__name__)
+        fbeta = FBeta(metric='precision', average=0)
+        self.assertEqual('precision_0', fbeta.__name__)
+        fbeta = FBeta(metric='recall', average='macro')
+        self.assertEqual('recall_macro', fbeta.__name__)
+        fbeta = FBeta(metric='recall', average='micro')
+        self.assertEqual('recall_micro', fbeta.__name__)
+        fbeta = FBeta(metric='recall', average=0)
+        self.assertEqual('recall_0', fbeta.__name__)
+        fbeta = FBeta(metric='fscore', average='macro', names='f')
+        self.assertEqual('f', fbeta.__name__)
```

Comment:
Add a case for a list of names
```suggestion
        self.assertEqual('f', fbeta.__name__)
        fbeta = FBeta(average='macro', names=['f', "f", "f"])
        self.assertEqual(["f", "f", "f"], fbeta.__name__)
```

Reply:
Pushed the same thing with different names instead.